src/metadataGeneration/customAttribute.ts

```diff
@@ -0,0 +1,67 @@
+import * as ts from 'typescript';
+import { getInitializerValue } from './initializer-value';
+import { Tsoa } from './tsoa';
+
+const getArgumentValue = (argument: ts.Expression) => {
```

Comment:
We could improve the structure here. If you don't switch on the kind but list 3 if blocks, each one will ensure type safety (by taking advantage of the type guards of the compiler API). 

I.e.:
```ts
if (ts.isStringLiteral(argument)) {
  return argument.text; // this is valid without casting now
}
```